analyses/new-feature-template.qmd

```diff
@@ -0,0 +1,1564 @@
+
+```{r, results = 'asis'}
+shap_tables[[25]]
+```
+
+###  `r township_names[26]`
+
+```{r, results = 'asis'}
+shap_tables[[26]]
+```
+
+###  `r township_names[27]`
+
+```{r, results = 'asis'}
+shap_tables[[27]]
+```
+
+###  `r township_names[28]`
+
+```{r, results = 'asis'}
+shap_tables[[28]]
+```
+
+###  `r township_names[29]`
+
+```{r, results = 'asis'}
+shap_tables[[29]]
+```
+
+###  `r township_names[30]`
+
+```{r, results = 'asis'}
+shap_tables[[30]]
+```
+
+###  `r township_names[31]`
+
+```{r, results = 'asis'}
+shap_tables[[31]]
+```
+###  `r township_names[32]`
+
+```{r, results = 'asis'}
+shap_tables[[32]]
+```
+
+###  `r township_names[33]`
+
+```{r, results = 'asis'}
+shap_tables[[33]]
+```
+
+###  `r township_names[34]`
+
+```{r, results = 'asis'}
+shap_tables[[34]]
+```
+
+###  `r township_names[35]`
+
+```{r, results = 'asis'}
+shap_tables[[35]]
+```
+
+###  `r township_names[36]`
+
+```{r, results = 'asis'}
+shap_tables[[36]]
+```
+
+###  `r township_names[37]`
+
+```{r, results = 'asis'}
+shap_tables[[37]]
+```
+
+###  `r township_names[38]`
+
+```{r, results = 'asis'}
+shap_tables[[38]]
+```
+
+:::
+
+## SHAP Comparison
+
+Here, we compare the SHAP values from the comparison data and the new data. The goal here is to see which other features, the added feature is influencing.
+```{r shap_comparisons}
+shap_comparison %>%
+  inner_join(shap, by = c("meta_card_num", "meta_pin")) %>%
+  select(-c(meta_pin, meta_card_num, meta_year.x, meta_year.y)) %>%
+  rename_with(~ sub("\\.x$", "_comp", .), ends_with(".x")) %>%
+  rename_with(~ sub("\\.y$", "_new", .), ends_with(".y")) %>%
+  mutate(across(ends_with("_new"), ~ . - get(sub("_new$", "_comp", cur_column())), .names = "{sub('_new$', '_diff', .col)}")) %>%
+  summarize(
+    across(ends_with("_diff"), list(
+      median_change = ~ median(.),
+      median_abs_change = ~ median(abs(.))
+    )),
+    across(ends_with("_comp"), list(median_comp = ~ median(.))),
+    across(ends_with("_new"), list(median_new = ~ median(.)))
+  ) %>%
+  pivot_longer(cols = everything(), names_to = c("feature", ".value"), names_sep = "_diff_|_comp_|_new_") %>%
+  arrange(feature) %>%
+  clean_column_values("feature") %>%
+  mutate(across(where(is.numeric), dollar, scale = 1, accuracy = 0.01)) %>%
+  datatable(
+    options = list(
+      scrollY = "300px",
+      scrollX = TRUE,
+      paging = FALSE,
+      searching = TRUE
+    ),
+    rownames = FALSE
+  )
+```
+
+
+## Correlation between SHAP and the Added Feature
+
+```{r shap_feature_correlation}
+correlation_value <- cor(pull(working_data_card, params$added_feature_shap), pull(working_data_card, params$added_feature), use = "complete.obs")
+```
+
+If the added feature leads to a significant change in predicted values, its correlation with the SHAP values should have a high absolute value. That correlation is `r round(correlation_value, 2)`.
+
+## Box Plot comparing SHAP and the Added Feature
+
+```{r shap_feature_plot}
```

Comment:
**suggestion:** There are pre-made packages for doing this that will make your life much easier. We're basically aiming to make [these plots](https://github.com/ModelOriented/shapviz/blob/main/man/figures/README-dep.png) from the `shapviz` package.

Reply:
These plots don't scale well to very large variable lists. We usually subset the 25 best indicators. Would it make sense to do the 10 + - from the variable we are looking at?

Reply:
I would make these for the top 10 features. For categoricals, you might be able to split up the categorical range into small multiples plots or similar.

Reply:
I'd assume that the vast majority of our new metrics won't be included in the top 10 features.